prometheus/gc_collector.go

```diff
@@ -0,0 +1,43 @@
+package prometheus
+
+import (
+	"runtime/debug"
+)
+
+type gcCollector struct {
+	numGC      Gauge
+	pauseTotal Gauge
+}
+
+// NewGCCollector returns a collector which exports metrics about the
+// garbage collection stats of the current process.
+func NewGCCollector() *gcCollector {
+	return &gcCollector{
+		numGC: NewGauge(GaugeOpts{
+			Name: "process_gc_num_total",
+			Help: "Total number of garbage collector incovations for the current process.",
```

Comment:
For comparison, what we do over in Java is https://github.com/prometheus/client_java/blob/master/simpleclient_hotspot/src/main/java/io/prometheus/client/hotspot/GarbageCollectorExports.java#L39